README.md

```diff
@@ -133,26 +133,73 @@ const App = () => {
 };
 ```
 
+### [Json Schema](http://json-schema.org/)
+
+The most standard way to validate JSON (implemented by [ajv](https://github.com/ajv-validator/ajv))
+
+[![npm](https://img.shields.io/bundlephobia/minzip/ajv@6.12.4?style=for-the-badge)](https://bundlephobia.com/result?p=ajv@6.12.4)
+
+```typescript jsx
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import { JSONSchema, jsonSchemaResolver } from '@hookform/resolvers';
```

Comment:
User should install 
`import Ajv, { ErrorObject, Options } from 'ajv';` at their app level, not within `resolves`.

Reply:
Ok! Of course.

Reply:
Thanks 👍

Reply:
Wait! `JSONSchema` is not from `ajv`.It's a type that I created. It need to be imported from the resolvers

Reply:
If the user want, they can also just pass the schema object. As long as it's correct, it'll pass the type check. This type is for users who want to statically type the schema object on declaration (and get auto completions from the IDE).

Reply:
As I've said somewhere in these conversations, the fact that ajv is used is just an implementation detail. The user will just pass the schema (hence the schema type (which is not defined in ajv)) and doesn't need to import ajv.

Reply:
we already having an issue with types in the lib, can we please above export external type from this lib? https://github.com/react-hook-form/resolvers/issues/20

Reply:
Ok, I read #20. I actually had already thought about that. I think this needs a restructuring.
I'd suggest having `@hookform/resolvers` defining the generic structure for validation resolvers, `@hookform/yupResolver` for the yup implementation, `@hookform/jsonSchemaResolver` for the JSON schema implementation, ... If all the implementations are under the same package then you'd have to install all the validation libraries as dependencies of this dependency just to use one of the implementations.

Reply:
So for now all the dependencies of each resolver implementation needs to be in the `package.json` so this is correct. Nevertheless it should be changed to different packages. One for the generic resolver logic (for custom resolvers, for example) and one for each other implementation in order to isolate the dependencies.